### ISSUE TITLE
MEMORY LEAK in onNotFound handler: repeatable MCVE

### ISSUE THREAD

**Issue by philbowles** [me-no-dev/ESPAsyncWebServer#528]
I'm raising another issue in the hope that someone might actually DO something, but this is the same problem as https://github.com/me-no-dev/ESPAsyncWebServer/issues/527 and https://github.com/me-no-dev/ESPAsyncWebServer/issues/521 which have received no attention. I have now narrowed it down further and provided an MCVE which shows the problem and is consistently repeatable.

```cpp
#include <ESP8266WiFi.h>
#include <ESPAsyncWebServer.h>

const char* ssid = "XXXXXXXX";
const char* password = "XXXXXXXX";

AsyncWebServer server(80);
void* grab;

void setup(void) {
  Serial.begin(115200);
  WiFi.begin(ssid, password);
  WiFi.waitForConnectResult();

  server.onNotFound([](AsyncWebServerRequest* request){   
    request->send(404, "text/plain","NO!");
  });
  
  server.begin();
  
  // grab 80% of the heap to speed things up
  grab=malloc((ESP.getFreeHeap() * 80) / 100);
  Serial.printf("LOW HEAP = %d\n",ESP.getFreeHeap());
}

void loop(void) {}
```

I then make repeated browser requests from 192.168.1.20 while causing my amazon echo dot on 192.168.1.8 to make simultaneous requests. On a low heap, it doesn't take long to show the problem. I have added 3 diagnostic lines to the library, whose content should be obvious:

```
LOW HEAP = 9664
ESPAsyncWebServer server started
T=15743 FH=8200 _server.onClient 3fffa094 192.168.1.20
T=15744 FH=7656 AsyncWebServerRequest 3fffa174 CTOR sizeof=284
T=15745 FH=7528 _server.onClient 3fffa294 192.168.1.20
T=15748 FH=6984 AsyncWebServerRequest 3fffa374 CTOR sizeof=284
T=15801 FH=6488 AsyncWebServerRequest DTOR 3fffa374
T=16037 FH=7520 _server.onClient 3fffa294 192.168.1.20
T=16038 FH=6976 AsyncWebServerRequest 3fffa374 CTOR sizeof=284
T=16159 FH=6488 AsyncWebServerRequest DTOR 3fffa174
T=16703 FH=7520 _server.onClient 3fffa01c 192.168.1.20
T=16704 FH=6976 AsyncWebServerRequest 3fffa0fc CTOR sizeof=284
T=16825 FH=6488 AsyncWebServerRequest DTOR 3fffa374
T=17183 FH=7520 _server.onClient 3fffa294 192.168.1.20
T=17184 FH=6976 AsyncWebServerRequest 3fffa374 CTOR sizeof=284
T=17249 FH=3896 _server.onClient 3fffb084 192.168.1.8
T=17250 FH=3352 AsyncWebServerRequest 3fffb164 CTOR sizeof=284
T=17303 FH=4336 AsyncWebServerRequest DTOR 3fffa0fc
T=17668 FH=2616 _server.onClient 3fffa01c 192.168.1.20
T=17669 FH=2072 AsyncWebServerRequest 3fffa6ec CTOR sizeof=284
T=17721 FH=4336 AsyncWebServerRequest DTOR 3fffa374
T=20993 FH=5872 AsyncWebServerRequest DTOR 3fffa6ec
T=27263 FH=5872 _server.onClient 3fffade4 192.168.1.20
T=27264 FH=5328 AsyncWebServerRequest 3fffaec4 CTOR sizeof=284
T=27266 FH=5152 _server.onClient 3fffa664 192.168.1.20
T=27268 FH=4608 AsyncWebServerRequest 3fffa744 CTOR sizeof=284
T=27321 FH=4168 AsyncWebServerRequest DTOR 3fffaec4
T=27762 FH=5200 _server.onClient 3fffade4 192.168.1.8
T=27763 FH=4656 AsyncWebServerRequest 3fffaec4 CTOR sizeof=284
T=28040 FH=408 _server.onClient 3fff9ff4 192.168.1.20

Exception (29):
epc1=0x40204195 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000

ctx: sys 
sp: 3fffec10 end: 3fffffb0 offset: 01a0

>>>stack>>>
3fffedb0:  38323d54 20303430 343d4846 5f203830  
3fffedc0:  76726573 6f2e7265 696c436e 20746e65  
3fffedd0:  66666633 34666639 32393120 3836312e  
3fffede0:  322e312e 00000a30 3fffee50 40206a7b  
3fffedf0:  3fffee30 3fffee00 00000008 40206aad  
3fffee00:  3ffef37c 0000179d 0000179d 4010020c  
3fffee10:  3ffef16c 00000198 3fffee50 00006d88  
3fffee20:  3ffef16c 00000198 0000011c 401004e4  
3fffee30:  2e323931 2e383631 0000011c 00006d88  
3fffee40:  3ffef16c 00000000 3fff9ff4 402054d5  
3fffee50:  00000000 00000000 00000000 40207ae8  
3fffee60:  1401a8c0 00000030 0000000b 3ffef61c  
3fffee70:  3fffaa0c 3ffef16c 000000d8 3ffef61c  
3fffee80:  3fff9ff4 3ffef16c 3fffaa0c 40205e8a  
3fffee90:  00000000 00000000 3fffaa0c 402060a4  
3fffeea0:  3fffaa0c 3ffef600 3ffef601 402060d0 
... etc
```

Interestingly, if I remove the server.onNotFound , I can't make it happen. I also cannot make it happen, no matter how fast I hit "refresh" if I only send in requests from a single IP.

With the explicit handler code included and using two IPs, I  can make it happen within seconds, every time...(within many minutes unless I shrink the heap first, but the same behaviour exists an all sizes of heap.)

The preconditions for the memory leak condition seems to be:
1) An explicit handler
2) simultaneous requests from DIFFERENT IPs

I have spent literally WEEKS trying to isolate this problem, I don't know what more I can do...I know the repo owner has other things in his life than this, but PLEASE PLEASE PLEASE can someone at least TRY to look at this and fix it?

**Comment by GeorgeFlorian** [comment]
Don't bother !
@me-no-dev Said he knows about this but that he either has no idea how to fix it or that he doesn't find the time.

And he isn't been finding the time for the past year or so.

https://github.com/me-no-dev/ESPAsyncWebServer/issues/529

**Comment by me-no-dev** [comment]
I have been finding time... no result yet. If anything it seems that 8266 has followed ESP32's issues. They both use the same LwIP... could be a clue. TBH any time I need a server, i use Async, just to give it more chance for resolving.

**Comment by Aircoookie** [comment]
@me-no-dev thank you for looking into it! From your comment it seems like you believe this issue is caused by upstream (either AsyncTCP, Arduino core or LWIP) and not by the library itself, right?

On an unrelated note, please merge https://github.com/me-no-dev/ESPAsyncWebServer/pull/491 :)

@philbowles What Arduino core and LwIP versions have you been using? If the results vary with the core used, that might give us more clues. This issue also seems to affects my projects so I'd be keen on seeing a fix!

**Comment by me-no-dev** [comment]
I think that the issue generally lies in some change in LwIP between the versions. Older versions behaved differently. While I was thinking that maybe RTOS is the issue on ESP32 (I mean my code is the issue but RTOS is the driving factor), the fact that same reports started appearing on 8266 kinda points towards that...

The issue is not in the AsyncWebServer lib at all. It's in AsyncTCP :)

**Comment by baggior** [comment]
> @me-no-dev 

> On an unrelated note, please merge #491 :)

**Comment by me-no-dev** [comment]
merged and also AsyncTCP has been updated. Please give a try :)

**Comment by Aircoookie** [comment]
@me-no-dev Awesome, thanks for the (hopefully) fix! I've never really had issues with ESP32, might be due to the larger amount of memory not showing such leaks as early as with ESP8266.

Can this fix also be applied to ESPAsyncTCP (8266 version) or do you suspect the issue on that platform has a different cause? I would be happy to help, but the TCP library code is beyond my comprehension :)

**Comment by me-no-dev** [comment]
the ESP32 issue was in multithreading. ESP8266 is not multithreaded, so that will need further investigation. TBH I have not booted 8266 in a loong time... :) will try soon.

**Comment by me-no-dev** [comment]
@philbowles can you also make it happen if you do not starve the memory as much? Like give it another 4-5K? Adding the callback and using different hosts will cause memory usage on their own.

**Comment by Aircoookie** [comment]
@me-no-dev In my application there is a `/freeheap` page (initial memory available after boot 27kB). If I poll that once per second while having an AsyncMqttClient connection to `broker.hivemq.com` established (not even publishing anything or getting messages send to the subscribed topic), the memory decreases by around 3kB about every 15 seconds (in one go, the 15 seconds in between the number stays the same). Until, at around 3-6kB left, it wdt resets.

I believe it might be due to some connection not being closed in time or a more specific issue caused by the MQTT library (in my case) or the Echo (in philbowles case). I tried the above MCVE (just commented out the memory starvation) and I am unable to get it to leak the heap with 2 hosts sending 10 req/sec each (tried both lwip 1 and 2). It went down by 20kB at one time but then recovered to the initial 45kB ten seconds later. I can exclude weak wifi signal as a possible cause, the router is 1m away and no interfering networks around.

**Comment by me-no-dev** [comment]
So since yesterday I have been testing camera streaming over Async. 25FPS+ for over 24hours. Not a byte lost. I randomly connected from another browser to loath things extra. No glitch at all :)

**Comment by Aircoookie** [comment]
Alright, I spoke to soon about the MCVE. It did crash, but only after 30 min. Before that, no leak occured, but then the heap rapidly declined over 10 sec and the system crashed. It happened again and again directly after the crash (due to built-up unhandled requests during the reset i suppose)
Good to hear that the streaming worked though, I will try and see if there anything else could be causing the issue.

<details><summary>Here is the modified MCVE</summary><p>

```cpp
//Arduino 1.8.9, tested with ESP8266 core v2.5.2 Lwip 1 and 2

#include <ESP8266WiFi.h>
#include <ESPAsyncWebServer.h>

const char* ssid = "x";
const char* password = "x";

AsyncWebServer server(80);

void setup(void) {
  Serial.begin(115200);
  WiFi.begin(ssid, password);
  WiFi.waitForConnectResult();
  Serial.println(WiFi.localIP());

  server.onNotFound([](AsyncWebServerRequest* request){   
    request->send(404, "text/plain","NO!");
    Serial.printf("LOW HEAP = %d\n",ESP.getFreeHeap());
  });
  
  server.begin();
}

void loop(void) {}
```

</p></details>

<details><summary>And the serial out</summary><p>

```
//was happily sitting at around 43kB for half an hour then:
LOW HEAP = 43720
LOW HEAP = 45304
LOW HEAP = 43720
LOW HEAP = 45584
LOW HEAP = 43720
LOW HEAP = 45480
LOW HEAP = 43720
LOW HEAP = 45320
LOW HEAP = 42776
LOW HEAP = 45320
LOW HEAP = 44320
LOW HEAP = 45320
LOW HEAP = 46248
LOW HEAP = 45920
LOW HEAP = 43704
LOW HEAP = 43720
LOW HEAP = 45584
LOW HEAP = 43720
LOW HEAP = 45464
LOW HEAP = 43720
LOW HEAP = 45320
LOW HEAP = 43720
LOW HEAP = 45320
LOW HEAP = 44320
LOW HEAP = 45320
LOW HEAP = 44320
LOW HEAP = 45320
LOW HEAP = 46248
LOW HEAP = 45272
LOW HEAP = 43040
LOW HEAP = 42120
LOW HEAP = 45584
LOW HEAP = 43392
LOW HEAP = 45320
LOW HEAP = 44320
LOW HEAP = 45296
LOW HEAP = 43992
LOW HEAP = 45320
LOW HEAP = 46248
LOW HEAP = 45320
LOW HEAP = 44968
LOW HEAP = 43720
LOW HEAP = 44968
LOW HEAP = 43720
LOW HEAP = 45464
LOW HEAP = 44320
LOW HEAP = 45320
LOW HEAP = 42776
LOW HEAP = 45320
LOW HEAP = 44648
LOW HEAP = 45320
LOW HEAP = 43704
LOW HEAP = 45320
LOW HEAP = 46248
LOW HEAP = 43720
LOW HEAP = 45912
LOW HEAP = 43720
LOW HEAP = 44984
LOW HEAP = 42776
LOW HEAP = 45320
LOW HEAP = 44320
LOW HEAP = 43720
LOW HEAP = 43392
LOW HEAP = 45320
LOW HEAP = 43376
LOW HEAP = 46528
LOW HEAP = 43984
LOW HEAP = 43720
LOW HEAP = 40840
LOW HEAP = 43648
LOW HEAP = 43056
LOW HEAP = 42456
LOW HEAP = 40856
LOW HEAP = 38664
LOW HEAP = 42784
LOW HEAP = 42464
LOW HEAP = 43328
LOW HEAP = 42096
LOW HEAP = 42792
LOW HEAP = 41192
LOW HEAP = 42792
LOW HEAP = 40848
LOW HEAP = 43392
LOW HEAP = 41456
LOW HEAP = 43056
LOW HEAP = 43056
LOW HEAP = 43056
LOW HEAP = 41456
LOW HEAP = 42968
LOW HEAP = 42656
LOW HEAP = 41984
LOW HEAP = 41976
LOW HEAP = 41296
LOW HEAP = 41904
LOW HEAP = 41240
LOW HEAP = 41232
LOW HEAP = 40560
LOW HEAP = 40224
LOW HEAP = 38944
LOW HEAP = 39552
LOW HEAP = 39216
LOW HEAP = 38552
LOW HEAP = 38208
LOW HEAP = 37872
LOW HEAP = 37184
LOW HEAP = 37200
LOW HEAP = 35920
LOW HEAP = 36176
LOW HEAP = 35856
LOW HEAP = 35168
LOW HEAP = 34576
LOW HEAP = 36208
LOW HEAP = 34464
LOW HEAP = 32720
LOW HEAP = 31792
LOW HEAP = 29864
LOW HEAP = 29072
LOW HEAP = 27328
LOW HEAP = 25136
LOW HEAP = 24472
LOW HEAP = 23520
LOW HEAP = 22272
LOW HEAP = 20080
LOW HEAP = 20960
LOW HEAP = 18752
LOW HEAP = 16112
LOW HEAP = 14368
LOW HEAP = 13688
LOW HEAP = 12440
LOW HEAP = 12240
LOW HEAP = 10520
LOW HEAP = 8920
LOW HEAP = 7496
LOW HEAP = 5776
LOW HEAP = 3608
LOW HEAP = 1416
LOW HEAP = 1080
LOW HEAP = 488

Abort called

>>>stack>>>

ctx: sys
sp: 3fffecc0 end: 3fffffb0 offset: 01b0
3fffee70:  40000f3d 00000023 3fff65c4 40204ca3  
3fffee80:  3fff65c4 3ffeed98 3ffeed99 40204ce8  
3fffee90:  ffffffff 3fffc6fc 00000001 40213726  
3fffeea0:  0000003c 401002d8 00000014 0000003c  
3fffeeb0:  3ffeef20 bb010a0a 0000004c 40207c78  
3fffeec0:  40210000 0000003c 3fffef30 3ffeef20  
3fffeed0:  3ffef014 3fff203c 3fff205a 40218759  
3fffeee0:  00000014 00000000 3ffeef20 40207a34  
3fffeef0:  3fffdc80 3ffef534 3fff38ec 3ffef5c4  
3fffef00:  00000008 3ffeef20 3fff203c 40210689  
3fffef10:  3fffdc80 3ffef534 3fff38ec 402104bf  
3fffef20:  4022f76e 3ffef534 3fff38ec 4022f77f  
3fffef30:  3fff204c 3fff203c 00000000 3ffe8510  
3fffef40:  4022a687 00000000 3fff38ec 402313e7  
3fffef50:  40000f49 3fffdab0 3fffdab0 40000f49  
3fffef60:  40000e19 40001878 00000002 00000000  
3fffef70:  3fffff10 aa55aa55 0000008c 401042f8  
3fffef80:  401042fe 00000002 00000000 73706368  
3fffef90:  4010000d 63612064 6864006b 20737063  
3fffefa0:  00000000 3fffef3c 00000000 3fffff48  
3fffefb0:  3fffffc0 00000000 00000000 feefeffe  
3fffefc0:  feefeffe feefeffe feefeffe feefeffe  
3fffefd0:  feefeffe feefeffe feefeffe feefeffe  
3fffefe0:  feefeffe feefeffe feefeffe feefeffe  
3fffeff0:  feefeffe feefeffe feefeffe feefeffe  
3ffff000:  feefeffe feefeffe feefeffe feefeffe  
3ffff010:  feefeffe feefeffe feefeffe feefeffe  
3ffff020:  feefeffe feefeffe feefeffe feefeffe  
3ffff030:  feefeffe feefeffe feefeffe feefeffe  
3ffff040:  feefeffe feefeffe feefeffe feefeffe  
3ffff050:  feefeffe feefeffe feefeffe feefeffe  
3ffff060:  feefeffe feefeffe feefeffe feefeffe  
3ffff070:  feefeffe feefeffe feefeffe feefeffe  
3ffff080:  feefeffe feefeffe feefeffe feefeffe  
3ffff090:  feefeffe feefeffe feefeffe feefeffe  
3ffff0a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff100:  feefeffe feefeffe feefeffe feefeffe  
3ffff110:  feefeffe feefeffe feefeffe feefeffe  
3ffff120:  feefeffe feefeffe feefeffe feefeffe  
3ffff130:  feefeffe feefeffe feefeffe feefeffe  
3ffff140:  feefeffe feefeffe feefeffe feefeffe  
3ffff150:  feefeffe feefeffe feefeffe feefeffe  
3ffff160:  feefeffe feefeffe feefeffe feefeffe  
3ffff170:  feefeffe feefeffe feefeffe feefeffe  
3ffff180:  feefeffe feefeffe feefeffe feefeffe  
3ffff190:  feefeffe feefeffe feefeffe feefeffe  
3ffff1a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff200:  feefeffe feefeffe feefeffe feefeffe  
3ffff210:  feefeffe feefeffe feefeffe feefeffe  
3ffff220:  feefeffe feefeffe feefeffe feefeffe  
3ffff230:  feefeffe feefeffe feefeffe feefeffe  
3ffff240:  feefeffe feefeffe feefeffe feefeffe  
3ffff250:  feefeffe feefeffe feefeffe feefeffe  
3ffff260:  feefeffe feefeffe feefeffe feefeffe  
3ffff270:  feefeffe feefeffe feefeffe feefeffe  
3ffff280:  feefeffe feefeffe feefeffe feefeffe  
3ffff290:  feefeffe feefeffe feefeffe feefeffe  
3ffff2a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff300:  feefeffe feefeffe feefeffe feefeffe  
3ffff310:  feefeffe feefeffe feefeffe feefeffe  
3ffff320:  feefeffe feefeffe feefeffe feefeffe  
3ffff330:  feefeffe feefeffe feefeffe feefeffe  
3ffff340:  feefeffe feefeffe feefeffe feefeffe  
3ffff350:  feefeffe feefeffe feefeffe feefeffe  
3ffff360:  feefeffe feefeffe feefeffe feefeffe  
3ffff370:  feefeffe feefeffe feefeffe feefeffe  
3ffff380:  feefeffe feefeffe feefeffe feefeffe  
3ffff390:  feefeffe feefeffe feefeffe feefeffe  
3ffff3a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff400:  feefeffe feefeffe feefeffe feefeffe  
3ffff410:  feefeffe feefeffe feefeffe feefeffe  
3ffff420:  feefeffe feefeffe feefeffe feefeffe  
3ffff430:  feefeffe feefeffe feefeffe feefeffe  
3ffff440:  feefeffe feefeffe feefeffe feefeffe  
3ffff450:  feefeffe feefeffe feefeffe feefeffe  
3ffff460:  feefeffe feefeffe feefeffe feefeffe  
3ffff470:  feefeffe feefeffe feefeffe feefeffe  
3ffff480:  feefeffe feefeffe feefeffe feefeffe  
3ffff490:  feefeffe feefeffe feefeffe feefeffe  
3ffff4a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff500:  feefeffe feefeffe feefeffe feefeffe  
3ffff510:  feefeffe feefeffe feefeffe feefeffe  
3ffff520:  feefeffe feefeffe feefeffe feefeffe  
3ffff530:  feefeffe feefeffe feefeffe feefeffe  
3ffff540:  feefeffe feefeffe feefeffe feefeffe  
3ffff550:  feefeffe feefeffe feefeffe feefeffe  
3ffff560:  feefeffe feefeffe feefeffe feefeffe  
3ffff570:  feefeffe feefeffe feefeffe feefeffe  
3ffff580:  feefeffe feefeffe feefeffe feefeffe  
3ffff590:  feefeffe feefeffe feefeffe feefeffe  
3ffff5a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff600:  feefeffe feefeffe feefeffe feefeffe  
3ffff610:  feefeffe feefeffe feefeffe feefeffe  
3ffff620:  feefeffe feefeffe feefeffe feefeffe  
3ffff630:  feefeffe feefeffe feefeffe feefeffe  
3ffff640:  feefeffe feefeffe feefeffe feefeffe  
3ffff650:  feefeffe feefeffe feefeffe feefeffe  
3ffff660:  feefeffe feefeffe feefeffe feefeffe  
3ffff670:  feefeffe feefeffe feefeffe feefeffe  
3ffff680:  feefeffe feefeffe feefeffe feefeffe  
3ffff690:  feefeffe feefeffe feefeffe feefeffe  
3ffff6a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff700:  feefeffe feefeffe feefeffe feefeffe  
3ffff710:  feefeffe feefeffe feefeffe feefeffe  
3ffff720:  feefeffe feefeffe feefeffe feefeffe  
3ffff730:  feefeffe feefeffe feefeffe feefeffe  
3ffff740:  feefeffe feefeffe feefeffe feefeffe  
3ffff750:  feefeffe feefeffe feefeffe feefeffe  
3ffff760:  feefeffe feefeffe feefeffe feefeffe  
3ffff770:  feefeffe feefeffe feefeffe feefeffe  
3ffff780:  feefeffe feefeffe feefeffe feefeffe  
3ffff790:  feefeffe feefeffe feefeffe feefeffe  
3ffff7a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff800:  feefeffe feefeffe feefeffe feefeffe  
3ffff810:  feefeffe feefeffe feefeffe feefeffe  
3ffff820:  feefeffe feefeffe feefeffe feefeffe  
3ffff830:  feefeffe feefeffe feefeffe feefeffe  
3ffff840:  feefeffe feefeffe feefeffe feefeffe  
3ffff850:  feefeffe feefeffe feefeffe feefeffe  
3ffff860:  feefeffe feefeffe feefeffe feefeffe  
3ffff870:  feefeffe feefeffe feefeffe feefeffe  
3ffff880:  feefeffe feefeffe feefeffe feefeffe  
3ffff890:  feefeffe feefeffe feefeffe feefeffe  
3ffff8a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff900:  feefeffe feefeffe feefeffe feefeffe  
3ffff910:  feefeffe feefeffe feefeffe feefeffe  
3ffff920:  feefeffe feefeffe feefeffe feefeffe  
3ffff930:  feefeffe feefeffe feefeffe feefeffe  
3ffff940:  feefeffe feefeffe feefeffe feefeffe  
3ffff950:  feefeffe feefeffe feefeffe feefeffe  
3ffff960:  feefeffe feefeffe feefeffe feefeffe  
3ffff970:  feefeffe feefeffe feefeffe feefeffe  
3ffff980:  feefeffe feefeffe feefeffe feefeffe  
3ffff990:  feefeffe feefeffe feefeffe feefeffe  
3ffff9a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9f0:  feefeffe feefeffe feefeffe feefeffe  
3ffffa00:  feefeffe feefeffe feefeffe feefeffe  
3ffffa10:  feefeffe feefeffe feefeffe feefeffe  
3ffffa20:  feefeffe feefeffe feefeffe feefeffe  
3ffffa30:  feefeffe feefeffe feefeffe feefeffe  
3ffffa40:  feefeffe feefeffe feefeffe feefeffe  
3ffffa50:  feefeffe feefeffe feefeffe feefeffe  
3ffffa60:  feefeffe feefeffe feefeffe feefeffe  
3ffffa70:  feefeffe feefeffe feefeffe feefeffe  
3ffffa80:  feefeffe feefeffe feefeffe feefeffe  
3ffffa90:  feefeffe feefeffe feefeffe feefeffe  
3ffffaa0:  feefeffe feefeffe feefeffe feefeffe  
3ffffab0:  feefeffe feefeffe feefeffe feefeffe  
3ffffac0:  feefeffe feefeffe feefeffe feefeffe  
3ffffad0:  feefeffe feefeffe feefeffe feefeffe  
3ffffae0:  feefeffe feefeffe feefeffe feefeffe  
3ffffaf0:  feefeffe feefeffe feefeffe feefeffe  
3ffffb00:  feefeffe feefeffe feefeffe feefeffe  
3ffffb10:  feefeffe feefeffe feefeffe feefeffe  
3ffffb20:  feefeffe feefeffe feefeffe feefeffe  
3ffffb30:  feefeffe feefeffe feefeffe feefeffe  
3ffffb40:  feefeffe feefeffe feefeffe feefeffe  
3ffffb50:  feefeffe feefeffe feefeffe feefeffe  
3ffffb60:  feefeffe feefeffe feefeffe feefeffe  
3ffffb70:  feefeffe feefeffe feefeffe feefeffe  
3ffffb80:  feefeffe feefeffe feefeffe feefeffe  
3ffffb90:  feefeffe feefeffe feefeffe feefeffe  
3ffffba0:  feefeffe feefeffe feefeffe feefeffe  
3ffffbb0:  00000005 00000000 00000020 40100df2  
3ffffbc0:  3ffe9015 401041ef 3ffeca78 feefeffe  
3ffffbd0:  40101b0d 3ffeca78 feefeffe feefeffe  
3ffffbe0:  00007fff 0000f478 3ffed4e8 40101ce2  
3ffffbf0:  3ffe989c 00000000 00000000 feefeffe  
3ffffc00:  00007fff 0000f478 40102122 00000100  
3ffffc10:  7fffffff 3ffe989c 3ffe989c 00000001  
3ffffc20:  00000001 00000080 feefeffe feefeffe  
3ffffc30:  feefeffe 0000f478 00000000 4000050c  
3ffffc40:  00000000 00000000 0000001f 40104ad1  
3ffffc50:  4000050c feefeffe feefeffe feefeffe  
3ffffc60:  400043d0 00000030 00000016 ffffffff  
3ffffc70:  400044ab 3fffc718 3ffffd60 08000000  
3ffffc80:  60000200 08000000 00000003 00000000  
3ffffc90:  00000005 00000000 0000000a 00000000  
3ffffca0:  00000005 00000000 0000000a 00000000  
3ffffcb0:  00000000 00000000 0000001f 40104ad1  
3ffffcc0:  4000050c 00000000 0000001f 40104ad1  
3ffffcd0:  40100dec 00000030 0000001f fffffffe  
3ffffce0:  00000000 00000000 0000001f 40104ad1  
3ffffcf0:  00000000 00000000 0000001f 40104ad1  
3ffffd00:  4000050c 00000000 0000001f 40104ad1  
3ffffd10:  40101a76 00000030 0000001f fffffffe  
3ffffd20:  00000000 00000000 0000001f 40104ad1  
3ffffd30:  4000050c 3ffe98cc 3ffed4e8 00000005  
3ffffd40:  40102075 00000030 0000001f fffffffe  
3ffffd50:  00000000 00000000 00000001 80000000  
3ffffd60:  3fffdab0 00000000 00015f7a 00000000  
3ffffd70:  00000005 00000000 00030000 00000100  
3ffffd80:  7fffffff 3ffe98c0 3ffe98c0 00000030  
3ffffd90:  c0070640 00000000 00000001 00000100  
3ffffda0:  00000064 00000001 4010413a 3ffed1d8  
3ffffdb0:  3ffed160 00000000 00000001 00000100  
3ffffdc0:  00000005 00000000 00000020 40100df2  
3ffffdd0:  3ffe9015 401041ef 3ffec9d8 000000ff  
3ffffde0:  00000002 00000000 00000020 40100df2  
3ffffdf0:  3ffe9012 401041d4 3ffec708 40101ce2  
3ffffe00:  00000005 00000000 00000020 40100df2  
3ffffe10:  3ffe9015 401041ef 3ffeca50 40100df2  
3ffffe20:  40101b0d 3ffeca50 3ffeca50 00000001  
3ffffe30:  00007fff 2cfdd97c 3ffed4e8 40101ce2  
3ffffe40:  3ffe989c 00000000 00000000 40101ce2  
3ffffe50:  00007fff 2cfdd97c 40102122 00000100  
3ffffe60:  7fffffff 3ffe989c 3ffe989c 00000001  
3ffffe70:  00000001 00004288 00040000 00000022  
3ffffe80:  3ffed1d8 2cfdd97c 00002200 4000050c  
3ffffe90:  3fffc278 40101ebc 3fffc200 00000022  
3ffffea0:  3ffe98cc 40101ebc 3fffc200 00000022  
3ffffeb0:  40206581 00000030 00000010 ffffffff  
3ffffec0:  40100565 00000000 00000001 00000001  
3ffffed0:  3fffdab0 00000000 3fffd9d0 3ffeea88  
3ffffee0:  00000000 00000000 00000001 3ffeea58  
3ffffef0:  00000000 3fffdad0 3ffeea88 00000030  
3fffff00:  00000000 3fffdad0 3ffeea88 00000030  
3fffff10:  3ffeeb28 00000142 00000142 402107ca  
3fffff20:  00000000 3ffee8b8 3ffefad4 402108cc  
3fffff30:  3fffff70 3fffff40 00000008 3ffeea88  
3fffff40:  3fffdad0 3ffee8b8 0000c558 00001853  
3fffff50:  00033fc0 00000001 3ffee8b8 40206fc4  
3fffff60:  3ffe858b 4020ff7e 3fffff40 3fffff70  
3fffff70:  40205024 3ffee8b8 3ffee8b8 3ffeea88  
3fffff80:  3fffdad0 3ffee8b8 3ffee968 40201154  
3fffff90:  00000000 00000000 00000001 402064e9  
3fffffa0:  3fffdad0 00000000 3ffeea58 4020657e  
<<<stack<<<

last failed alloc call: 4020FF7E(216)

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v8b899c12
~ld
10.10.1.187
LOW HEAP = 50344
LOW HEAP = 47848
LOW HEAP = 46200
LOW HEAP = 42968
LOW HEAP = 44456
LOW HEAP = 42800
LOW HEAP = 43840
LOW HEAP = 41624
LOW HEAP = 43224
LOW HEAP = 43496
LOW HEAP = 41768
LOW HEAP = 41768
LOW HEAP = 42040
LOW HEAP = 39496
LOW HEAP = 41480
LOW HEAP = 39880
LOW HEAP = 40480
LOW HEAP = 38400
LOW HEAP = 39464
LOW HEAP = 36920
LOW HEAP = 37848
LOW HEAP = 38792
LOW HEAP = 37792
LOW HEAP = 37768
LOW HEAP = 37336
LOW HEAP = 37336
LOW HEAP = 36664
LOW HEAP = 35992
LOW HEAP = 35024
LOW HEAP = 34376
LOW HEAP = 35320
LOW HEAP = 34648
LOW HEAP = 34624
LOW HEAP = 33976
LOW HEAP = 33416
LOW HEAP = 31352
LOW HEAP = 32632
LOW HEAP = 31032
LOW HEAP = 31744
LOW HEAP = 29472
LOW HEAP = 31400
LOW HEAP = 29128
LOW HEAP = 29560
LOW HEAP = 29944
LOW HEAP = 27784
LOW HEAP = 28328
LOW HEAP = 28600
LOW HEAP = 27656
LOW HEAP = 28040
LOW HEAP = 27576
LOW HEAP = 27256
LOW HEAP = 26584
LOW HEAP = 26584
LOW HEAP = 23480
LOW HEAP = 24968
LOW HEAP = 25240
LOW HEAP = 25016
LOW HEAP = 25624
LOW HEAP = 25288
LOW HEAP = 24624
LOW HEAP = 24504
LOW HEAP = 23616
LOW HEAP = 22552
LOW HEAP = 22216
LOW HEAP = 22936
LOW HEAP = 22600
LOW HEAP = 21320
LOW HEAP = 21976
LOW HEAP = 20528
LOW HEAP = 18808
LOW HEAP = 16920
LOW HEAP = 14256
LOW HEAP = 12536
LOW HEAP = 11968
LOW HEAP = 13504
LOW HEAP = 13496
LOW HEAP = 10808
LOW HEAP = 9088
LOW HEAP = 14176
LOW HEAP = 14096
LOW HEAP = 13760
LOW HEAP = 14368
LOW HEAP = 11792
LOW HEAP = 8704
LOW HEAP = 6984
LOW HEAP = 5264
LOW HEAP = 4600
LOW HEAP = 5792
LOW HEAP = 5104
LOW HEAP = 4792
LOW HEAP = 6584
LOW HEAP = 6016
LOW HEAP = 3984
LOW HEAP = 2128
LOW HEAP = 848
LOW HEAP = 1128
LOW HEAP = 2040
LOW HEAP = 2312
LOW HEAP = 1360

Abort called

>>>stack>>>

ctx: sys
sp: 3fffecc0 end: 3fffffb0 offset: 01b0
3fffee70:  3fffdab0 00000000 3fff2ebc 40204ca3  
3fffee80:  3fff2ebc 3ffeed98 3ffeed99 40204ce8  
3fffee90:  00000000 3fffdad0 3ffeea88 40213726  
3fffeea0:  0000003c 401002d8 00000014 0000003c  
3fffeeb0:  3ffeef20 bb010a0a 0000004c 40207c78  
3fffeec0:  40210000 0000003c 3fffef30 3ffeef20  
3fffeed0:  3ffef014 3fff735c 3fff737a 40218759  
3fffeee0:  00000014 00000000 3ffeef20 40207a34  
3fffeef0:  3fffdc80 3ffef534 3fff04d4 3ffef5c4  
3fffef00:  00000008 3ffeef20 3fff735c 40210689  
3fffef10:  3fffdc80 3ffef534 3fff04d4 402104bf  
3fffef20:  4022f76e 3ffef534 3fff04d4 4022f77f  
3fffef30:  3fff736c 3fff735c 00000000 3ffe8510  
3fffef40:  4022a687 00000000 3fff04d4 402313e7  
3fffef50:  40000f49 3fffdab0 3fffdab0 40000f49  
3fffef60:  40000e19 40001878 00000002 3fffffb0  
3fffef70:  3fffff10 aa55aa55 0000008c 401042f8  
3fffef80:  401042fe 00000002 3fffffb0 73706368  
3fffef90:  4010000d 63612064 6864006b 20737063  
3fffefa0:  00000000 3fffef3c 00000000 3fffff48  
3fffefb0:  3fffffc0 00000000 00000000 feefeffe  
3fffefc0:  feefeffe feefeffe feefeffe feefeffe  
3fffefd0:  feefeffe feefeffe feefeffe feefeffe  
3fffefe0:  feefeffe feefeffe feefeffe feefeffe  
3fffeff0:  feefeffe feefeffe feefeffe feefeffe  
3ffff000:  feefeffe feefeffe feefeffe feefeffe  
3ffff010:  feefeffe feefeffe feefeffe feefeffe  
3ffff020:  feefeffe feefeffe feefeffe feefeffe  
3ffff030:  feefeffe feefeffe feefeffe feefeffe  
3ffff040:  feefeffe feefeffe feefeffe feefeffe  
3ffff050:  feefeffe feefeffe feefeffe feefeffe  
3ffff060:  feefeffe feefeffe feefeffe feefeffe  
3ffff070:  feefeffe feefeffe feefeffe feefeffe  
3ffff080:  feefeffe feefeffe feefeffe feefeffe  
3ffff090:  feefeffe feefeffe feefeffe feefeffe  
3ffff0a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff0f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff100:  feefeffe feefeffe feefeffe feefeffe  
3ffff110:  feefeffe feefeffe feefeffe feefeffe  
3ffff120:  feefeffe feefeffe feefeffe feefeffe  
3ffff130:  feefeffe feefeffe feefeffe feefeffe  
3ffff140:  feefeffe feefeffe feefeffe feefeffe  
3ffff150:  feefeffe feefeffe feefeffe feefeffe  
3ffff160:  feefeffe feefeffe feefeffe feefeffe  
3ffff170:  feefeffe feefeffe feefeffe feefeffe  
3ffff180:  feefeffe feefeffe feefeffe feefeffe  
3ffff190:  feefeffe feefeffe feefeffe feefeffe  
3ffff1a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff1f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff200:  feefeffe feefeffe feefeffe feefeffe  
3ffff210:  feefeffe feefeffe feefeffe feefeffe  
3ffff220:  feefeffe feefeffe feefeffe feefeffe  
3ffff230:  feefeffe feefeffe feefeffe feefeffe  
3ffff240:  feefeffe feefeffe feefeffe feefeffe  
3ffff250:  feefeffe feefeffe feefeffe feefeffe  
3ffff260:  feefeffe feefeffe feefeffe feefeffe  
3ffff270:  feefeffe feefeffe feefeffe feefeffe  
3ffff280:  feefeffe feefeffe feefeffe feefeffe  
3ffff290:  feefeffe feefeffe feefeffe feefeffe  
3ffff2a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff2f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff300:  feefeffe feefeffe feefeffe feefeffe  
3ffff310:  feefeffe feefeffe feefeffe feefeffe  
3ffff320:  feefeffe feefeffe feefeffe feefeffe  
3ffff330:  feefeffe feefeffe feefeffe feefeffe  
3ffff340:  feefeffe feefeffe feefeffe feefeffe  
3ffff350:  feefeffe feefeffe feefeffe feefeffe  
3ffff360:  feefeffe feefeffe feefeffe feefeffe  
3ffff370:  feefeffe feefeffe feefeffe feefeffe  
3ffff380:  feefeffe feefeffe feefeffe feefeffe  
3ffff390:  feefeffe feefeffe feefeffe feefeffe  
3ffff3a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff3f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff400:  feefeffe feefeffe feefeffe feefeffe  
3ffff410:  feefeffe feefeffe feefeffe feefeffe  
3ffff420:  feefeffe feefeffe feefeffe feefeffe  
3ffff430:  feefeffe feefeffe feefeffe feefeffe  
3ffff440:  feefeffe feefeffe feefeffe feefeffe  
3ffff450:  feefeffe feefeffe feefeffe feefeffe  
3ffff460:  feefeffe feefeffe feefeffe feefeffe  
3ffff470:  feefeffe feefeffe feefeffe feefeffe  
3ffff480:  feefeffe feefeffe feefeffe feefeffe  
3ffff490:  feefeffe feefeffe feefeffe feefeffe  
3ffff4a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff4f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff500:  feefeffe feefeffe feefeffe feefeffe  
3ffff510:  feefeffe feefeffe feefeffe feefeffe  
3ffff520:  feefeffe feefeffe feefeffe feefeffe  
3ffff530:  feefeffe feefeffe feefeffe feefeffe  
3ffff540:  feefeffe feefeffe feefeffe feefeffe  
3ffff550:  feefeffe feefeffe feefeffe feefeffe  
3ffff560:  feefeffe feefeffe feefeffe feefeffe  
3ffff570:  feefeffe feefeffe feefeffe feefeffe  
3ffff580:  feefeffe feefeffe feefeffe feefeffe  
3ffff590:  feefeffe feefeffe feefeffe feefeffe  
3ffff5a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff5f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff600:  feefeffe feefeffe feefeffe feefeffe  
3ffff610:  feefeffe feefeffe feefeffe feefeffe  
3ffff620:  feefeffe feefeffe feefeffe feefeffe  
3ffff630:  feefeffe feefeffe feefeffe feefeffe  
3ffff640:  feefeffe feefeffe feefeffe feefeffe  
3ffff650:  feefeffe feefeffe feefeffe feefeffe  
3ffff660:  feefeffe feefeffe feefeffe feefeffe  
3ffff670:  feefeffe feefeffe feefeffe feefeffe  
3ffff680:  feefeffe feefeffe feefeffe feefeffe  
3ffff690:  feefeffe feefeffe feefeffe feefeffe  
3ffff6a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff6f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff700:  feefeffe feefeffe feefeffe feefeffe  
3ffff710:  feefeffe feefeffe feefeffe feefeffe  
3ffff720:  feefeffe feefeffe feefeffe feefeffe  
3ffff730:  feefeffe feefeffe feefeffe feefeffe  
3ffff740:  feefeffe feefeffe feefeffe feefeffe  
3ffff750:  feefeffe feefeffe feefeffe feefeffe  
3ffff760:  feefeffe feefeffe feefeffe feefeffe  
3ffff770:  feefeffe feefeffe feefeffe feefeffe  
3ffff780:  feefeffe feefeffe feefeffe feefeffe  
3ffff790:  feefeffe feefeffe feefeffe feefeffe  
3ffff7a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff7f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff800:  feefeffe feefeffe feefeffe feefeffe  
3ffff810:  feefeffe feefeffe feefeffe feefeffe  
3ffff820:  feefeffe feefeffe feefeffe feefeffe  
3ffff830:  feefeffe feefeffe feefeffe feefeffe  
3ffff840:  feefeffe feefeffe feefeffe feefeffe  
3ffff850:  feefeffe feefeffe feefeffe feefeffe  
3ffff860:  feefeffe feefeffe feefeffe feefeffe  
3ffff870:  feefeffe feefeffe feefeffe feefeffe  
3ffff880:  feefeffe feefeffe feefeffe feefeffe  
3ffff890:  feefeffe feefeffe feefeffe feefeffe  
3ffff8a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff8f0:  feefeffe feefeffe feefeffe feefeffe  
3ffff900:  feefeffe feefeffe feefeffe feefeffe  
3ffff910:  feefeffe feefeffe feefeffe feefeffe  
3ffff920:  feefeffe feefeffe feefeffe feefeffe  
3ffff930:  feefeffe feefeffe feefeffe feefeffe  
3ffff940:  feefeffe feefeffe feefeffe feefeffe  
3ffff950:  feefeffe feefeffe feefeffe feefeffe  
3ffff960:  feefeffe feefeffe feefeffe feefeffe  
3ffff970:  feefeffe feefeffe feefeffe feefeffe  
3ffff980:  feefeffe feefeffe feefeffe feefeffe  
3ffff990:  feefeffe feefeffe feefeffe feefeffe  
3ffff9a0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9b0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9c0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9d0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9e0:  feefeffe feefeffe feefeffe feefeffe  
3ffff9f0:  feefeffe feefeffe feefeffe feefeffe  
3ffffa00:  feefeffe feefeffe feefeffe feefeffe  
3ffffa10:  feefeffe feefeffe feefeffe feefeffe  
3ffffa20:  feefeffe feefeffe feefeffe feefeffe  
3ffffa30:  feefeffe feefeffe feefeffe feefeffe  
3ffffa40:  feefeffe feefeffe feefeffe feefeffe  
3ffffa50:  feefeffe feefeffe feefeffe feefeffe  
3ffffa60:  feefeffe feefeffe feefeffe feefeffe  
3ffffa70:  feefeffe feefeffe feefeffe feefeffe  
3ffffa80:  feefeffe feefeffe feefeffe feefeffe  
3ffffa90:  feefeffe feefeffe feefeffe feefeffe  
3ffffaa0:  feefeffe feefeffe feefeffe feefeffe  
3ffffab0:  feefeffe feefeffe feefeffe feefeffe  
3ffffac0:  feefeffe feefeffe feefeffe feefeffe  
3ffffad0:  feefeffe feefeffe feefeffe feefeffe  
3ffffae0:  feefeffe feefeffe feefeffe feefeffe  
3ffffaf0:  feefeffe feefeffe feefeffe feefeffe  
3ffffb00:  feefeffe feefeffe feefeffe feefeffe  
3ffffb10:  feefeffe feefeffe feefeffe feefeffe  
3ffffb20:  feefeffe feefeffe feefeffe feefeffe  
3ffffb30:  feefeffe feefeffe feefeffe feefeffe  
3ffffb40:  feefeffe feefeffe feefeffe feefeffe  
3ffffb50:  feefeffe feefeffe feefeffe feefeffe  
3ffffb60:  feefeffe feefeffe feefeffe feefeffe  
3ffffb70:  feefeffe feefeffe feefeffe feefeffe  
3ffffb80:  feefeffe feefeffe feefeffe feefeffe  
3ffffb90:  feefeffe feefeffe feefeffe feefeffe  
3ffffba0:  feefeffe feefeffe feefeffe feefeffe  
3ffffbb0:  3ffe9015 401041ef 3ffeca78 feefeffe  
3ffffbc0:  40101b0d 3ffeca78 feefeffe feefeffe  
3ffffbd0:  00007fff 0000b97d 3ffed4e8 40101ce2  
3ffffbe0:  3ffe98c0 00000000 00000000 feefeffe  
3ffffbf0:  00007fff 0000b97d 40102122 00000100  
3ffffc00:  7fffffff 3ffe98c0 3ffe98c0 00000001  
3ffffc10:  00000001 00004288 feefeffe feefeffe  
3ffffc20:  00000000 0000b97d 00000000 4000050c  
3ffffc30:  3fffc278 40101ebc 3fffc200 00000022  
3ffffc40:  3ffe98cc 401033f7 3ffed190 feefeffe  
3ffffc50:  40237137 00000030 00000016 ffffffff  
3ffffc60:  40238717 3ffee7ac 60000e54 00000054  
3ffffc70:  00028002 d14223a0 3ffee7b8 00000003  
3ffffc80:  00000009 fffff000 ffffffbf 3ffee748  
3ffffc90:  00000005 00000000 0000000a 00000000  
3ffffca0:  00000005 00000000 0000000a 00000000  
3ffffcb0:  3ffe9015 401041ef 3ffeca50 00000022  
3ffffcc0:  00000000 a0000000 00000000 0000001c  
3ffffcd0:  00005000 00000000 00005000 00000000  
3ffffce0:  00000000 00000000 0000001f 40104ad1  
3ffffcf0:  4000050c 3ffffde3 4022b7e3 00000001  
3ffffd00:  40102ba1 00000030 0000001f fffffffe  
3ffffd10:  3ffecaa0 00000008 00000000 3ffe9890  
3ffffd20:  00000001 001c1da3 00007fff 0000012b  
3ffffd30:  3ffed4e8 00000006 00000000 3ffe9890  
3ffffd40:  00000001 3ffe9890 3ffed4e8 00000030  
3ffffd50:  00000005 0003d291 40102122 00000001  
3ffffd60:  00000000 00000001 3ffffe40 4020e562  
3ffffd70:  3ffffde3 3ffe8598 00000000 3ffedb3c  
3ffffd80:  3ffe8c3c 0003d291 00000000 3ffe8598  
3ffffd90:  00000000 3ffe8599 3ffffe40 4020e74c  
3ffffda0:  00000000 ffffffff ffffffff 00000000  
3ffffdb0:  00000005 00000000 00302064 00000100  
3ffffdc0:  00000005 00000000 00000020 40100df2  
3ffffdd0:  3ffe9015 401041ef 3ffecaa0 00000100  
3ffffde0:  00000002 00000000 00000020 40100df2  
3ffffdf0:  3ffe9012 401041d4 3ffec780 40100df2  
3ffffe00:  00000001 401033f7 3ffed1d8 40100df2  
3ffffe10:  4010385a 401041ef 3ffeca78 40101ce2  
3ffffe20:  40101b0d 3ffeca78 00000000 40101ce2  
3ffffe30:  4010333e 00000028 7fffffff 00000002  
3ffffe40:  00000128 00000000 00000000 00000001  
3ffffe50:  401021bf 00080000 40102122 00000100  
3ffffe60:  00000000 3ffe98a8 00002200 4000050c  
3ffffe70:  00000000 00000000 0000001f 40104ad1  
3ffffe80:  4000050c 0065afda 00002200 4000050c  
3ffffe90:  40000f83 00000030 0000001b ffffffff  
3ffffea0:  40000ea3 00000023 00000001 00000001  
3ffffeb0:  3fffdab0 00000000 3fffd9d0 3ffeea88  
3ffffec0:  00000000 00000000 00000001 3ffeea58  
3ffffed0:  00000000 3fffdad0 3ffeea88 00000030  
3ffffee0:  ffffffff 3fffc6fc 00000001 3ffeea58  
3ffffef0:  00000000 3fffdad0 3ffeea88 00000030  
3fffff00:  00000000 3fffdad0 3ffeea88 00000030  
3fffff10:  3ffeeb28 0000012b 0000012b 402107ca  
3fffff20:  00000000 3ffee8b8 3ffefb54 402108cc  
3fffff30:  3fffff70 3fffff40 00000008 3ffeea88  
3fffff40:  3fffdad0 3ffee8b8 0000c4a8 00001853  
3fffff50:  0001ab40 00000001 3ffee8b8 40206fc4  
3fffff60:  3ffe858b 4020ff7e 3fffff40 3fffff70  
3fffff70:  40205024 3ffee8b8 3ffee8b8 3ffeea88  
3fffff80:  3fffdad0 3ffee8b8 3ffee968 40201154  
3fffff90:  00000000 00000000 00000001 402064e9  
3fffffa0:  3fffdad0 00000000 3ffeea58 4020657e  
<<<stack<<<

last failed alloc call: 4020FF7E(216)

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v8b899c12
~ld
10.10.1.187
LOW HEAP = 50520
LOW HEAP = 47848
LOW HEAP = 46072
LOW HEAP = 46072
LOW HEAP = 45128
LOW HEAP = 44456
LOW HEAP = 44488
LOW HEAP = 43840
LOW HEAP = 43840
LOW HEAP = 40952
LOW HEAP = 43496
LOW HEAP = 40896
LOW HEAP = 41208
LOW HEAP = 42152
LOW HEAP = 38936
LOW HEAP = 40432
LOW HEAP = 40808
LOW HEAP = 40808
LOW HEAP = 39784
LOW HEAP = 39136
LOW HEAP = 37752
LOW HEAP = 37736
LOW HEAP = 38120
LOW HEAP = 38120
LOW HEAP = 37120
LOW HEAP = 36504
LOW HEAP = 36776
LOW HEAP = 34824
LOW HEAP = 33560
LOW HEAP = 34376
LOW HEAP = 35320
LOW HEAP = 32216
LOW HEAP = 33792
LOW HEAP = 34088
LOW HEAP = 34088
LOW HEAP = 33416
LOW HEAP = 32416
LOW HEAP = 30200
LOW HEAP = 31960
```

</p></details>

**Comment by philbowles** [comment]
@aircookie @me-no-dev Apologies for delay, and thanks for responses. I have tried all versions of lwip above 1.4 and also core 2.4.2 and 2.5.0 - same results.

"Alright, I spoke to soon about the MCVE. It did crash, but only after 30 min. Before that, no leak occured, but then the heap rapidly declined over 10 sec and the system crashed." I'm glad it's not just me, that's exactly what happens, "out of nowhere"

@me-no-dev I depleted the heap to speed up the error log. It will do it from any heap size, its just takes longer because the "overlap"that causes it is rare and loses "only" 2k, so if I start with 28k and the "overlap" occurs once every 1/2 hour, it takes 7hours to fail.

**Comment by cgmckeever** [comment]
> So since yesterday I have been testing camera streaming over Async. 25FPS+ for over 24hours. Not a byte lost. I randomly connected from another browser to loath things extra. No glitch at all :)

@me-no-dev do you have example code of this? I appear to be having a memory panic when I do `webSocket.binaryAll(buf, len);`
